src/protocol/primitives/nullable-string.ts

```diff
@@ -0,0 +1,21 @@
+import type { ReadBuffer, Serializable, WriteBuffer } from '../serialization';
+import { Int16 } from './int16';
+
+export class NullableString implements Serializable {
+  constructor(public readonly value: string | null) {}
+
+  public static deserialize(buffer: ReadBuffer): NullableString {
+    const length = Int16.deserialize(buffer);
+    if (length.value === -1) {
+      return new NullableString(null);
+    }
+    return new NullableString(buffer.readString(length.value));
+  }
+
+  public serialize(buffer: WriteBuffer): void {
+    new Int16(this.value === null ? -1 : this.value.length).serialize(buffer);
+    if (this.value !== null) {
```

Comment:
As there is already an explicit null-check, can we consider getting rid of the short-hand if else one line earlier?